text/text_blob.cpp

```diff
@@ -69,10 +69,12 @@ gfx::RectF TextBlob::RunInfo::getGlyphBounds(const size_t i) const
   if (bounds.isEmpty()) {
     FontMetrics metrics;
     font->metrics(&metrics);
-    bounds.w = metrics.avgCharWidth;
-    bounds.h = -metrics.capHeight;
+    // avgCharWidth can be 0, so we grab the next most useful thing, the height
+    bounds.w = metrics.avgCharWidth > 0 ? metrics.avgCharWidth : metrics.xHeight;
+    bounds.h = 1.0;
```

Comment:
I'm not sure about this one, `xHeight` is the height of the lowercase `x` character. It feels wrong to use this value as the whitespace width (and hard coding 1.0 as the height).

To test this we can use the `text_shape` which is a mini text editor, put the cursor at a whitespace to test the whitespace bounds.

We should be able to set bounds.w = the advance in X axis for whitespaces, but I'm not sure where to find this value on Skia.

Reply:
Yeah this one's the most contentious change, the hardcoded 1 was to try and fix a specific bug of things being too tall in some circumstances, it should not stay like this. I need to look into more of the documentation

Reply:
I'm trying a new approach with just the normal " " character glyph bounds and it looks to behave correctly - at least more consistently than attempting to use FontMetrics. I think the ideal solution would be to shape the entire set of glyphs in one go with skia using `getWidthsBounds` but that'd mean a full refactor of how the runs are working currently.